### PR TITLE
Add LDAP configuration

### PR DESCRIPTION
**Pull request by 4ludwig4**
Tested and worked.

Introduced an `ldap.enable` flag to easily enable ldap usage and not having to rely on some combination of set variables like in ansible which is not consistent.

### CODE REVIEW

File: artemis/Chart.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
+#TODO: figure out how and when this is updated
```
Comment (4ludwig4) [inline artemis/Chart.yaml:18]: I don't know the process of how we update chart versions or if we do this at all for the `develop` branch?

File: artemis/templates/configmaps/artemis-usermanagement-configmap.yml [hunk 2]
```diff
@@ -10,5 +10,11 @@ data:
   {{- else }}
   artemis.user-management.use-external: "false"
   {{- end }}
-
-
+  #TODO: is the if block okay like this? based on ansible template or already if just url for instance is set?
```
Comment (4ludwig4) [inline artemis/templates/configmaps/artemis-usermanagement-configmap.yml]: Not 100 percent sure if the logic makes sense. Used the ansible template as a base.

File: artemis/templates/configmaps/artemis-usermanagement-configmap.yml [hunk 3]
```diff
@@ -10,5 +10,11 @@ data:
...
   artemis.user-management.use-external: "false"
   {{- end }}
-
-
+  {{- with .Values.application.userManagement.ldap }}
+  {{- if .enabled }}
+  artemis.user-management.ldap.url: {{ required "LDAP URL is required" .url | quote }}
+  artemis.user-management.ldap.base: {{ required "LDAP BASE is required" .base | quote }}
+  artemis.user-management.ldap.allowed-username-pattern: {{ .allowedUsernamePattern | squote }}
+  management.health.ldap.enabled: "true"
```
Comment (4ludwig4) [inline artemis/templates/configmaps/artemis-usermanagement-configmap.yml:18]: Not sure if this should be separated or in this block. I don't know how this is usually done in helm chart projects. There are 2 ways to do this I guess. Either like I did it here to follow the features and set configs where most of the feature is configured or to follow the `yaml` configuration file and for this example create another file with another block checking if the `userManagement.ldap.enabled` is `true`.

**Review by sleiss — APPROVED** [review]
I've tested it and it works, but I sadly don't have experience regarding your questions, maybe @Mtze can help here?

File: artemis/templates/configmaps/artemis-usermanagement-configmap.yml (hunk 3 above)
Comment (Mtze) [reply artemis/templates/configmaps/artemis-usermanagement-configmap.yml:18]: I think your solution is fine for now :)

**Review by Mtze — APPROVED** [review]

**Pull request merged**